AK/FloatingPoint.h

```diff
@@ -258,6 +258,17 @@ constexpr double convert_to_native_double(I input) { return float_to_float<Doubl
 template<typename I>
 constexpr float convert_to_native_float(I input) { return float_to_float<SingleFloatingPointBits>(input).as_float(); }
 
+template<typename T>
+constexpr bool is_signaling_nan(T input)
+{
+    if (!isnan(input))
+        return false;
+
+    auto extract = FloatExtractor<T>(input);
+
+    return extract.exponent == FloatExtractor<T>::exponent_max && (extract.mantissa >> (FloatExtractor<T>::mantissa_bits - 1)) == 0;
```

Comment:
How about we also have a happy path for the future where `__builtin_issignaling` is available, i.e:

```cpp
#if __has_builtin(__builtin_issignaling)
    return __builtin_issignaling(input);
#else
    generic approach
#endif
```